pkg/cloudprovider/providers/clc/sdkCreds.go

```diff
@@ -0,0 +1,227 @@
+func (creds *implCreds) CredsLogin(username string, password string) error {
+	e := creds.CredsLogout(true)
+	if e != nil {
+		return e
+	}
+
+	if username == "" {
+		return clcError("cannot log in, username not provided")
+	} else if password == "" {
+		return clcError("cannot log in, password not provided")
+	}
+
+	body := fmt.Sprintf("{\"username\":\"%s\",\"password\":\"%s\"}", username, password)
+	b := bytes.NewBufferString(body)
+
+	authresp := AuthLoginResponseJSON{}
+
+	// but don't call invokeHTTP here, because we're a layer underneath that.
+	full_url := ("https://" + creds.AuthServer + creds.AuthURI)
+	req, err := http.NewRequest("POST", full_url, b)
+	if err != nil {
+		return clcError(fmt.Sprintf("could not create HTTP request: url=%s, err=%s", full_url, err.Error()))
+	}
+
+	req.Header.Add("Content-Type", "application/json") // incoming body to be a marshaled object already
+	req.Header.Add("Host", creds.AuthServer)           // the reason we take server and uri separately
+	req.Header.Add("Accept", "application/json")
+	// do not send Authorization - we are asking to become authorized
+	req.Header.Add("Connection", "close")
+
+	// this should be the normal code
+	resp, err := http.DefaultClient.Do(req) // execute the call
+
+	// instead, we have this which tolerates bad certs [fixme both here and in invokeHTTP]
+	// tlscfg := &tls.Config{InsecureSkipVerify: true} // true means to skip the verification
+	// transp := &http.Transport{TLSClientConfig: tlscfg}
+	// client := &http.Client{Transport: transp}
+	// resp, err := client.Do(req)
+	// end of tolerating bad certs.  Do not keep this code - it allows MITM etc. attacks
+
+	if err != nil { // failed HTTP call
+		return clcError(fmt.Sprintf("authorization call failed, err=%s", err.Error()))
+	}
+
+	if (resp.StatusCode < 200) || (resp.StatusCode >= 300) {
+		return clcError(fmt.Sprintf("authorization call failed, HTTP status=%d", resp.StatusCode))
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&authresp)
+	if err != nil {
+		return clcError(fmt.Sprintf("JSON decode failed, err=%s", err.Error()))
+	} // otherwise we now have an auth response in authresp
+
+	glog.Info(fmt.Sprintf("assigning new token, do this:  export CLC_API_TOKEN=%s\n", authresp.BearerToken))
+	glog.Info(fmt.Sprintf("also CLC_API_USERNAME=%s  CLC_API_ACCOUNT=%s  CLC_API_LOCATION=%s\n", authresp.Username, authresp.AccountAlias, authresp.LocationAlias))
+
+	creds.Username = authresp.Username
+	creds.Password = password
+	creds.AccountAlias = authresp.AccountAlias
+	creds.LocationAlias = authresp.LocationAlias
+	creds.BearerToken = authresp.BearerToken
+
+	return nil // no error
+}
+
+// whole clc package uses this, it's just here in creds because this is the bottom of the dependency stack
+func clcError(content string) error { // caller probably to use fmt.Sprintf to make the input string
+	if content == "" {
+		content = "<error text not available>"
+	}
+
+	full_err := "CLC: " + content
+
+	glog.Info(full_err)
```

Comment:
This should be glog.Error. but this is likely going to cause these errors to be called twice.